internal/awslambda/extension/manager_test.go

```diff
@@ -162,6 +162,21 @@ func EventNextHandler(tb testing.TB, ctx context.Context, statusCode, shutdownAf
 	})
 }
 
+func TelemetryHandler(tb testing.TB, statusCode int) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(tb, http.MethodPut, r.Method)
+		require.Contains(tb, r.Header, api.LambdaExtensionIdentifierHeaderKey)
+		require.Equal(tb, tb.Name(), r.Header.Get(api.LambdaExtensionIdentifierHeaderKey))
+		w.WriteHeader(statusCode)
+	})
+}
+
+func availableAddr() string {
+	l, _ := net.Listen("tcp", ":0")
+	defer l.Close()
+	return l.Addr().String()
+}
```

Comment:
Please use `httptest.NewServer` instead of trying to figure out which port is free.

Reply:
Might be related to [this](https://github.com/atlassian/gostatsd/pull/652#discussion_r1446874217) , but I use this for the telemetry server since it takes an address and it's in a separate package so we don't have access to the underlying server

How should I structure it here?

Reply:
Yes, you're correct. let me follow up on that comment there to help with context.

Reply:
I've done the refactor for server but I still need an addr lol

I'll save the full end to end test for the next PR when flushes are added in.